slsReceiverSoftware/src/MasterFileUtility.cpp

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: LGPL-3.0-or-other
+// Copyright (C) 2021 Contributors to the SLS Detector Package
+#include "MasterFileUtility.h"
+#include "sls/container_utils.h"
+
+#include <iomanip>
+
+namespace masterFileUtility {
+
+std::string CreateMasterBinaryFile(const std::string filePath,
+                                   const std::string fileNamePrefix,
+                                   const uint64_t fileIndex,
+                                   const bool overWriteEnable,
+                                   const bool silentMode,
+                                   MasterAttributes *attr) {
+    // create file name
+    std::ostringstream os;
+    os << filePath << "/" << fileNamePrefix << "_master"
+       << "_" << fileIndex << ".json";
+    std::string fileName = os.str();
+
+    // create file
+    FILE *fd{nullptr};
```

Comment:
This part looks a bit convoluted maybe you could rewrite to something like:
```
    FILE *fd{};
    if (!overWriteEnable)
        fopen(fileName.c_str(), "wx");
    else
        fd = fopen(fileName.c_str(), "w");
    
    if(!fd)
        //throw stuff

    return fd;
```

Or this form:

```
    std::string mode = "w";
    if (!overWriteEnable)
        mode = "wx";

    FILE *fd = fopen(fileName.c_str(), mode.c_str());
    if(!fd)
        //throw stuff
```

Reply:
done!